code/modules/surgery/bodyparts/species_parts/misc_bodyparts.dm

```diff
@@ -504,6 +516,18 @@
 	)
 	return ..()
 
+/obj/item/bodypart/arm/right/golem/set_owner(new_owner)
+	. = ..()
+	if (. == FALSE)
+		return
+	if (owner)
+		owner.AddComponentFrom(REF(src), /datum/component/shovel_hands)
+		return
```

Comment:
this shouldn't return, if there is an old owner they won't have the component removed